distribution/push_v2.go

```diff
@@ -425,26 +417,17 @@ func (pd *v2PushDescriptor) uploadUsingSession(
 	diffID layer.DiffID,
 	layerUpload distribution.BlobWriter,
 ) (distribution.Descriptor, error) {
-	arch, err := pd.layer.TarStream()
+	reader, release, err := pd.layer.Content(ctx, progressOutput, pd.ID(), schema2.MediaTypeLayer)
```

Comment:
Not sure if this is what you were originally suggesting, but I was thinking it might make sense to just have `Content` return the payload in the format it has (i.e. with compression if using a blob store, without compression if using `TarStream`), and indicate to the caller whether compression is applied. The push code would wrap the stream with a compressor if necessary. I think this would simplify the progress monitoring. Currently `storeLayer`'s `Content` method creates a progress reader with the hardcoded message `Pushing`, which doesn't feel right.